x-pack/legacy/plugins/maps/public/layers/sources/es_source.js

```diff
@@ -344,4 +345,25 @@ export class AbstractESSource extends AbstractVectorSource {
 
     return resp.aggregations;
   }
+
+  getValueSuggestions = async (fieldName, query) => {
```

Comment:
This does not work for IP-fields.

We should also booleans separately here. E.g. manually respond to `t*` or `f*` inputs.


It crashes now (e.g. test with `Cancelled`-field of `kibana_sample_flights`).

```
Uncaught (in promise) TypeError: searchValue.toLowerCase is not a function
    at getSelectedOptionForSearchValue (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:47184)
    at collectMatchingOption (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:47192)
    at :5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:47236
    at Array.forEach (<anonymous>)
    at getMatchingOptions (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:47219)
    at getDerivedStateFromProps (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:45286)
    at applyDerivedStateFromProps (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:362398)
    at updateClassInstance (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:362984)
    at updateClassComponent (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:366860)
    at beginWork$1 (:5601/ymj/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:368626)
```

Reply:
Looks like filter pill editor does something similar and does not provide type ahead for non-string fields. https://github.com/elastic/kibana/blob/master/src/plugins/data/public/ui/filter_bar/filter_editor/phrase_suggestor.tsx#L56. I can add something similiar.

```
  protected isSuggestingValues() {
    const shouldSuggestValues = this.services.uiSettings.get('filterEditor:suggestValues');
    const { field } = this.props;
    return shouldSuggestValues && field && field.aggregatable && field.type === 'string';
  }
```

Reply:
Debugging this further, suggestions for boolean will be possible.

Its not possible to get suggestions for ip field types because the autocomplete service users terms aggregation with `include` that uses a regular expression. This is not supported for ip field types.

Request generated by autocomplete service
```
GET kibana_sample_data_logs/_search
{
 "size": 0,
 "query": {
  "bool": {
   "filter": []
  }
 },
 "aggs": {
  "suggestions": {
   "terms": {
    "field": "clientip",
    "include": ".*",
    "execution_hint": "map",
    "shard_size": 10
   }
  }
 }
}
```

Response
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "aggregation_execution_exception",
        "reason" : "Aggregation [suggestions] cannot support regular expression style include/exclude settings as they can only be applied to string fields. Use an array of values for include/exclude clauses"
      }
    ],
    "type" : "search_phase_execution_exception",
    "reason" : "all shards failed",
    "phase" : "query",
    "grouped" : true,
    "failed_shards" : [
      {
        "shard" : 0,
        "index" : "kibana_sample_data_logs",
        "node" : "Td_Is2e9S-iJ8DroYKLkHA",
        "reason" : {
          "type" : "aggregation_execution_exception",
          "reason" : "Aggregation [suggestions] cannot support regular expression style include/exclude settings as they can only be applied to string fields. Use an array of values for include/exclude clauses"
        }
      }
    ]
  },
  "status" : 500
}
```

Reply:
+1 on discarding suggestions for IP-field